clm_demeter/prepare_baselayer.py

```diff
@@ -0,0 +1,161 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt # to generate plots
+from netCDF4 import Dataset     # to work with NetCDF files
+import os                       # operating system interface
+from scipy.interpolate import griddata
+
+__author__ = 'Min Chen and Eva Sinha'
+__email__ = 'min.chen@pnnl.gov and eva.sinha@pnnl.gov'
+__copyright__ = 'Copyright (c) 2017, Battelle Memorial Institute'
+
+# ------------------------------------------------------------------------
+def interpolate_coarse_fine(data, coarse_grid_coord, fine_grid_cooord):
+    """ Interpolate values of variable from coarse grid to fine grid
+    :param data:             subset(only land cells) of coarse grid coordinates along with values for variable
+    :param coarse_grid_coord pandas dataframe containing latitude and longitude coordinate of coarse grid
+    :param fine_grid_coord   pandas dataframe containing latitude and longitude coordinate of fine grid
+    
+    """
+    # Join the subset grid coordinate data with full coordinate data
+    coarse_grid_data = pd.merge(coarse_grid_coord, data, how='left')
+
+    # Interpolate from coarse grid to finer resolution grid. 
+    # Convert dataframe to ndarray for method='nearest' to work
+    data_fine = griddata(points=coarse_grid_data[['Latcoord','Loncoord']].values, 
+                         values=coarse_grid_data[data.columns[2]].values, 
+                         xi=fine_grid_coord.values, 
+                         method='nearest')
+    
+    return data_fine
+# ------------------------------------------------------------------------
+
+project_dir = '/pic/projects/im3/lulc_thrust_area/minchen/Demeter_0.05deg_trial/Demeter_0.05deg_trial'
+
+# Base layer file for Plant Function Type (PFT)
+# baselayer_pft_file = os.path.join(project_dir, 'mksrf_78pft_landuse_rc2000_c130927.nc')
+baselayer_pft_file = 'mksrf_78pft_landuse_rc2000_c130927.nc'
+
+# Open netCDF file
+dataset = Dataset(baselayer_pft_file, mode='r') # file handle, open in read only mode
+
+# Read latitude, longitude values
+lat = dataset.variables['LAT']  # dim = lat
+lon = dataset.variables['LON']  # dim = lon
+
+lat_xy = dataset.variables['LATIXY'] # dim = (lat, lon)
+lon_xy = dataset.variables['LONGXY'] # dim = (lat, lon)
+landmask = dataset.variables['LANDMASK'] # dim = (lat, lon)
+
+# Read data for PFT variable [pft (79), lat (3600), lon (7200)]
+pft = dataset.variables['PCT_PFT']
+
+close the file
+dataset.close()
+
+# flip the data in latitude so North Hemisphere is up on the plot
+# flipud will convert the netcdf variable to numpy array
+lat      = np.flipud(lat) 
+lat_xy   = np.flipud(lat_xy)
+lon_xy   = np.flipud(lon_xy)
+landmask = np.flipud(landmask)
+pft      = pft[:,::-1,:] 
+
+# PFT values are converted from percentage (0-100) to fraction (0-1)
+pft = pft*0.05*0.05*0.01; # unit conversion
```

Comment:
For Python, include spacing between operations (e.g., `*`).  Explain these multipliers and pass them as variables.